src/mapnode.cpp

```diff
@@ -170,64 +170,158 @@ void MapNode::rotateAlongYAxis(const NodeDefManager *nodemgr, Rotation rot)
 	}
 }
 
-void transformNodeBox(const MapNode &n, const NodeBox &nodebox,
-	const NodeDefManager *nodemgr, std::vector<aabb3f> *p_boxes,
-	u8 neighbors = 0)
-{
-	std::vector<aabb3f> &boxes = *p_boxes;
-
-	if (nodebox.type == NODEBOX_FIXED || nodebox.type == NODEBOX_LEVELED) {
-		const auto &fixed = nodebox.fixed;
-		int facedir = n.getFaceDir(nodemgr, true);
-		u8 axisdir = facedir >> 2;
-		facedir &= 0x03;
-
-		boxes.reserve(boxes.size() + fixed.size());
-		for (aabb3f box : fixed) {
-			if (nodebox.type == NODEBOX_LEVELED)
+void buildFixedNodeBox(const MapNode &n, const NodeBox &nodebox, const NodeDefManager *nodemgr,
+		std::vector<aabb3f> &boxes, std::vector<aabb3f> input_boxes,
+		enum NodeBoxType nbt) {
```

Comment:
line length and types
```suggestion
void buildFixedNodeBox(const MapNode &n, const NodeBox &nodebox,
		const NodeDefManager *nodemgr,
		std::vector<aabb3f> &boxes, const std::vector<aabb3f> &input_boxes,
		NodeBoxType nbt) {
```

Reply:
also why are we passing `nodebox` when it's not used?